### ISSUE TITLE
How to use with MSVC

### ISSUE THREAD

**Issue by caizhe666** [eshard/obfuscator-llvm#12]
My develop environment is Windows and Visual Studio, how can I use the project with them, I didn't see any documentation. Sincerely thanks to reply.

**Comment by MMitsuha** [comment]
It gives me a lot of errors...
I followed the README, and I failed to complie `obfuscator-llvm`

[build.log](https://github.com/eshard/obfuscator-llvm/files/12385137/build.log)

**Comment by PatriceBlin** [comment]
Most of the logs are warnings, on LLVM headers and some on our plugins.

But your error is 

```
C:\Users\cz251\source\repos\obfuscator-llvm\utils/CryptoUtils.h(85): fatal error C1189: #error: "Unknown endianness of the compilation platform, check this header aes_encrypt.h"
```

It's from

[./utils/CryptoUtils.h](https://github.com/eshard/obfuscator-llvm/blob/a67b09de57df5cfa8fbab831a511702119db8c2c/utils/CryptoUtils.h#L83)

```
#if !defined(ENDIAN_BIG) && !defined(ENDIAN_LITTLE)
#error                                                                         \
    "Unknown endianness of the compilation platform, check this header aes_encrypt.h"
#endif
```

If you are building from a Windows x86_64 could you try this modification ?


```diff
diff --git a/utils/CryptoUtils.h b/utils/CryptoUtils.h
index 8964c41..6cc9617 100644
--- a/utils/CryptoUtils.h
+++ b/utils/CryptoUtils.h
@@ -47,7 +47,7 @@ extern ManagedStatic<CryptoUtils> cryptoutils;
 #endif
 #define ENDIAN_64BITWORD
 
-#elif defined(__x86_64__)
+#elif defined(__x86_64__) || defined(_M_X64)
 
 #ifndef ENDIAN_LITTLE
 #define ENDIAN_LITTLE
```

I don't have a Windows with MSVC on hand to check if it works but I saw this macro used inside LLVM so it might just works.
